npm/private/npm_package.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+# --- begin runfiles.bash initialization v3 ---
+# Copy-pasted from the Bazel Bash runfiles library v3.
+set -uo pipefail; set +e; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+source "$0.runfiles/$f" 2>/dev/null || \
+source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+{ echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v3 ---
+
+readonly PNPM_PATH="$(rlocation pnpm/pnpm.sh)"
+readonly PACKAGE_DIR="$(rlocation {{PACKAGE_DIR}})"
+
+
+"${PNPM_PATH}" publish $PACKAGE_DIR --no-git-checks $@
```

Comment:
how does `pnpm publish` compare to `npm publish`?

Reply:
AFAIK, it performs some prechecks and delegates the work to `npm publish`.

Reply:
users transitively depend on npm.

Reply:
hmmm. how does it get npm? might not be hermetic here

Reply:
it most probably relies on PATH.

Reply:
you can get npm itself from the node toolchain. check what js_binary does what `include_npm` is set to True. https://docs.aspect.build/rules/aspect_rules_js/docs/js_binary#include_npm

Reply:
if `pnpm publish` doesn't add much value over `npm publish` then seems simpler to just call the latter here

Reply:
one thing I know `pnpm publish` will do is resolve `workspace:*` deps in the package.json of the package being published but I think that complexity can be dealt with in a future PR

Reply:
in npm_package case looks like it doesn't add much value.

Reply:
it's just a fancy wrapper that knows how to do pnpm-workspace multi-package releases etc.

Reply:
done. now uses js_binary instead.